### ISSUE TITLE
Updates, bugs, etc

### ISSUE THREAD

**Issue by fallengravity** [Asymetrex/desktop-app#3]
1.) This appears to be a bug - when you click trade, it directs you to a trading page but said page doesn't exist resulting in "Invalid Symbol" on the chart:
![image](https://user-images.githubusercontent.com/37632291/80865488-1db50700-8c8a-11ea-84e6-d2e99de4f19e.png)

2.) Another issue is that if you're not logged into the app - it fails to download the update, the moment you login then it updates without an issue. The zip file contains a recording of what I'm talking about.

[2020-05-02 15-26-54.zip](https://github.com/Asymetrex/desktop-app/files/4567798/2020-05-02.15-26-54.zip)

3.) I'm not sure if you just haven't implement this yet or if it's missing but once logged in and on a market to trade there are no buy/sell boxes. I assume this hasn't been implement yet as you haven't moved to api integration fully yet.

4.) On a trading page for the market there is a search bar, this search bar doesn't seem to do anything would it be possible to either add it to the markets page so the user can search for markets or to the withdraw/deposit pages.

5.) Please make sure the app saves the login key for the user - I forgot to mention this, it'd be ideal to encrypt it's saving as well otherwise the user will be required to regenerate a key on the website every time they logout as the private key is only displayed when generated on the website, never again after that. Also it should have two keys: ACCESS_KEY & SECRET_KEY.

6.) In regard to i18n support - your implementation is perfect.

7.) Installer Image: 
[aeximageapp.zip](https://github.com/Asymetrex/desktop-app/files/4567825/aeximageapp.zip)

**Comment by fallengravity** [comment]
8.) Also just noticed the prices on the trading page are infinitely loading, they never appear, unsure if this is a bug or not implemented yet: 
![image](https://user-images.githubusercontent.com/37632291/80866004-b0ef3c00-8c8c-11ea-9902-d7be6ac6dc00.png)

**Comment by fallengravity** [comment]
[icons.zip](https://github.com/Asymetrex/desktop-app/files/4567852/icons.zip)

**Comment by newCodeRunner** [comment]
1) Trading Page fallback set to 'monbtc' otherwise it will load first market tab's first currency. if you want different, please let me know - Done

2) There is an open bug in the dep, however, i have checked and implemented a work around, should be working now - Done

3) not implemented yet - need access/secret code to implement that - Pending

4) The search wasn't implemented, now it is, also added to Markets Page.
On withdraw/deposit each market is to be shown or each currency? will implement search according to that - meanwhile Done for both Markets and Trade Pages

5) Do we need 1 or both keys to log in? Currently added both and saving both and requiring both to log in using node native module keytar: https://www.npmjs.com/package/keytar which uses native OS password storage so OS will handle the security for us. However, if user wants to remove saved codes, they should just log out, if they want app to remember they should just quit, added a confirm prompt explaining that- Done

6) i18n- thank you - No further action - Done

7) Installer image - should be in the same color scheme and file size as in the src/assets folder. I tried to do it online, but it destroyed the image quality, please check and provide again (hint you can take and edit the src/assets/app-installer-sidebar-example.bmp file for that - it should have all rights specs pre-configured)- Done with Poor image quality - Pending

8) the prices values are not being retrieved in the api .tickerslist(callback, false), is this usual and i have to take another api for prices? Or will it only update after log In? - Pending
eg. result is an obejct with keys like but there is no price property in ticker object

```
{
ltcbtc: {
at: 1588497095,
ticker: {
buy: "0.0",
sell: "0.0",
low: "0.0",
high: "0.0",
last: "0.1",
vol: "0.0",
volbtc: "0",
volquote: 0,
change: "0.0",
name: "BTC/LTC",
currency: "Litecoin",
}
}
```


9) Coins images - Done

Questions from My Side
10) Have you seen notifications? when would you want me to app to display them to user?

11) How frequently does the coin data change in markets table? do we really need updates every 2 seconds, or can we safely extend the limit? how long max do you recommend, if yes?

**Comment by newCodeRunner** [comment]
12) There is one image missing : 
808 Coin in DOGE market

**Comment by fallengravity** [comment]
1.) The search for deposit/withdraw pages shouldn't work off the markets but instead via name, so that the user can quickly find and make a withdrawal/deposit eg:
![Screenshot 2020-05-03 at 15 45 22](https://user-images.githubusercontent.com/37632291/80915856-1f97cc80-8d55-11ea-8286-8cd306e12a55.png)

5.) The user should utilize both their public + private key. The confirmations sound good.

7.) I will get the installer image sorted out in the next week or to, you can proceed with how it is currently until I am able to get you a new version.

8.) This was my bad the "last" property in the tickers list object is the price value.

10.) The notifications are great, there should be notifications when a withdraw is made containing withdraw details - when I buy/sell trade is made containing the details such as amount, value, and coin. Also, would there be a way for us as the exchange to send push notifications to the User and have them saved in the app so we could inform them of maintenance, changes, etc??

11.) I think we can safely move it to 120 seconds, for the time being, once there is a larger user base this will need to be decreased.

12.) 808 Coin is in the process of being removed from the platform.

**Comment by fallengravity** [comment]
I will send you keys in a short bit.

Changes:

1.) Please can the main table load in based on volume so that coins with the high volume are listed at the top and then after it's the coins with 0 volume.

2.) Please can you add a small spacing between the currency icon and the name on the main table.

3.) Please add the currency icons to withdraw/deposit when you get to that part.

**Comment by newCodeRunner** [comment]
Earlier Pending Points
1. Buy and Sell Boxes - which api do i need to call for that? this portion is a little unclear, can you please explain? i can not understand from the api. - Pending

2. Search in all pages complete - Done

3. Installer Image - waiting - Pending

4. Prices not appearing fixed - Done

5. Auto updating of data shifted to 120 secs - Done

6. Search according to currency name in withdraw/deposit - Done

7. Login using public and private keys - Done

8. Notifications on buy/trade pending since it is linked to point number 1 - Pending

New Issues/requests

1. Sorting of table according to volume - Done

2. Add small gap between names and icons - Done

3. Currency Icons to withdraw and deposit - Done

**Comment by fallengravity** [comment]
You should use the .createOrder method. 

EG:

API.createOrder('ethobtc', 'buy', 100, 0.00000001)
API.createOrder('ethobtc', 'sell', 100, 0.00000001)

**Comment by newCodeRunner** [comment]
And what should the table display by default? when the user is logged in?

**Comment by newCodeRunner** [comment]
Moreover, Citadel Coin is missing, can you please provide that?

also, i havent implemented the api callback failures when user is offline, or when the request is unsuccessful, this is noted and will be implemented.

**Comment by dylie** [comment]
1) Note while I was testing the withdraw page:
When you for example fill in "xerom" address + amount, it doesn't do the actual withdraw but it asks to fill in more "required" even if it's another coin (see screenshot)
![Screenshot_7](https://user-images.githubusercontent.com/38665633/81471630-c61b1c00-91f2-11ea-8788-08c69db2f690.jpg)

2) Filling in random words for the login box seem to log you in aswel. Is it possible to display an error message when the api key / login key is invalid?

3) Can you make the total owned coins and locked coins a bit bigger so it's easier to see?
![Screenshot_9](https://user-images.githubusercontent.com/38665633/81471680-10040200-91f3-11ea-8b5a-36acd8549860.jpg)

Beside that, great work!

**Comment by fallengravity** [comment]
![coin-i-ctl](https://user-images.githubusercontent.com/37632291/81472174-86563380-91f6-11ea-8a6a-93db8d5b5356.png)

**Comment by fallengravity** [comment]
The table display as-is is good.

**Comment by newCodeRunner** [comment]
1. Buy and Sell Boxes - implemented auto refreshing on 12 seconds as well. which is the field for amount? Also have I selected the right api? I am using .orderBook api to get current buy and sell orders for the boxes. Notifications on buy/sell i.e. createOrder api transactions have not been implemented yet  - Pending

2. Installer Image - waiting - Pending

3. Withdraw method is still not implemented - Pending

4. Random entry login blocked, error message appearing - Done

5. Make coin balance larger - Done

6. Add CTL coin - Done

7. Some server busy errors resolved when loading addresses on Deposits Page. Also made app update data sooner on last failed to get data due to server busy etc on markets, trades and deposits page. - Done

**Comment by fallengravity** [comment]
API.createOrder('market', 'buy/sell', amount, price)

![Screenshot 2020-05-10 at 11 51 11](https://user-images.githubusercontent.com/37632291/81495985-90d70280-92b4-11ea-82b9-3d28246930b2.png)

in this case, it'll be:

Amount: 100
Price: 0.0000001
Total: Amount * price
Market: ethobtc (example)
method: buy/sell

**Comment by fallengravity** [comment]
1.) Balance sizes: please increase the font-size to 16px - image attached
![Screenshot 2020-05-10 at 12 01 39](https://user-images.githubusercontent.com/37632291/81496184-05f70780-92b6-11ea-8482-4eb330f4c178.png)

2.) Is it possible to cache the wallet addresses? They almost never change so if we could cache them for say 72 hours that'd be great, maybe add a refresh button so the user can refresh their addresses inside those 72 hours.

3.) Getting a few loading errors it doesn't recover from instead you have to click the homepage and then go back to where you were to get the data to load:
![Screenshot 2020-05-10 at 12 04 19](https://user-images.githubusercontent.com/37632291/81496242-9afa0080-92b6-11ea-8e75-8de76bd70fcc.png)

**Comment by fallengravity** [comment]
4.) Installer Image - Don't worry about this at all. We will sort it out once your work is done.

5.) Buy/sell is correct using order API amount column is price * volume

**Comment by newCodeRunner** [comment]
1. Notifications + Buy/Sell Orders - Pending
2. Amount prop fixed in Buy/Sell Tables - Done
3. Random Entry Login Protection - Done
4. Loading errors fixed now - Done
5. Font size to 16px - Done (earlier was 14px)
6. Installer Image -Noted

7. createOrder api - Pending
8. Withdraw api - Pending

9. About Google-Analytics - one solution could be opening an invisible window/iframe in background redirecting to a page/link with only analytics scripts and no HTML, this window url can be changed in background as user navigates in app without user feeling any lags. But this would mean some new URLS to be hosted by your server - Pending

10. Custom Notifications - we can use a similar solution to 9, or if you can extend a function in api at your backend, i can create a timeout and check for notifications at regular intervals -Pending

11. Going to try and build on linux machine, want to remove all bugs before making the code for transactions due to limited balance avail to me - Pending

**Comment by newCodeRunner** [comment]
12. Also about saving deposit addresses to 72 hours, i am already calling the addresses only once, and they are being saved in memory, i can cache them locally until user log outs? the refresh button can also be added, no issue - Pending

**Comment by fallengravity** [comment]
1.) Custom Notifications - I will write a Custom REST API for this and I will send it to you as soon as it's done.

2.) Google-Analytics - That's okay we can live without it.

3.) Deposit Addresses - Cache until log out is perfect no need for a 72-hour cache then however the refresh button would be very useful, thanks.

**Comment by fallengravity** [comment]
Asymetrex Message API for custom notifications: 

https://messages.asymetrex.com/messages

**Comment by newCodeRunner** [comment]
1. Notifications + Buy/Sell Orders - Pending
2. createOrder api - Pending
3. Withdraw api - Pending
4. Custom Notifications, please just make sure id remains number and always increases in your api - it is filtering with ids greater than last cached - Done
5. Caching Deposit Addresses - Done
6. Linux and Mac Build Configuration have been updated - Testing Required - (need your support with mac, my linux machine was out of date, need some time with it) - Pending

**Comment by fallengravity** [comment]
All looks good (Except Bug posted on MacOS issue)

Please can you limit the amount column on the trade page to 8 Decimal places .toFixed(8) is fine Please can you also add to the Amount Column name the Base pair & volume column the target pair as shown in the example below:

![Screenshot 2020-05-11 at 10 39 30](https://user-images.githubusercontent.com/37632291/81541645-b840d480-9373-11ea-97a1-aca2ab119b52.png)

**Comment by newCodeRunner** [comment]
1. Notifications + Buy/Sell Orders - Pending
2. createOrder api - Pending
3. Withdraw api - Pending
4. Mac Build Config Updated again - Linux Testing Required - Pending
5. 8 Decimal places - Done
6. Base Pair and Volume Pair - Done

**Comment by dylie** [comment]
Hey mate,

I think I got the installer image to work now... It's the same bithdepth (4) now as the example. It should work... If it doesn't, don't bother =D

**Comment by dylie** [comment]
[app-installer-sidebar.zip](https://github.com/Asymetrex/desktop-app/files/4610006/app-installer-sidebar.zip)

**Comment by newCodeRunner** [comment]
![image](https://user-images.githubusercontent.com/57590297/81642535-0ff74280-943d-11ea-96cc-52dc0f155f78.png)

1. Installer Image - Done
2. Linux Testing looks okay, there were some missing icons, however, i have resolved them too, going to push a commit - Done

Some notes for Linux:
AppImage will be built as only these support the auto update features at the moment.
If you recommend users use appImageLauncher on linux then they would be able to integrate it on desktop (showing on launch menu with asymetrex icon) otherwise, the users will see AppImage icon on the folder where they downloaded the app, and they can double click it to run the app; However, once the app is launched, i have confirmed it is showing the asymetrex icon in taskbar and in notifications on Ubuntu 18

Btw any updates regarding Mac Icon Issue, i closed the issue but am still waiting for feedback, dont personally want the terminal icon to appear on the app =D

**Comment by fallengravity** [comment]
Hey, awesome, yes MacOS is fixed and working perfectly. Thanks for letting me know about Linux 👍

**Comment by newCodeRunner** [comment]
Everything seems in order on Linux

![App Opened](https://user-images.githubusercontent.com/57590297/81669524-c835e280-945f-11ea-9bad-2d86437a6135.png)
![Downloaded Folder](https://user-images.githubusercontent.com/57590297/81669530-c9670f80-945f-11ea-9a24-3228921e4fa1.png)
![First-Time-Use (appImageLauncherInstalled)](https://user-images.githubusercontent.com/57590297/81669532-c9ffa600-945f-11ea-85b5-1312154355ec.png)
![Integrated with AppImage (Shows in Launch Menu)Screenshot from 2020-05-12 14-34-32](https://user-images.githubusercontent.com/57590297/81669534-ca983c80-945f-11ea-9270-1d16799b53d9.png)
![Notifications](https://user-images.githubusercontent.com/57590297/81669537-cb30d300-945f-11ea-9929-d09bf5e08c44.png)
![Trade Page on Linux](https://user-images.githubusercontent.com/57590297/81669538-cbc96980-945f-11ea-8b8c-c29137fd3c8b.png)

**Comment by newCodeRunner** [comment]
Opening a new issue for the pending tasks: (this takes forever to refresh and scroll down)

1. Notifications + Buy/Sell Orders - Pending
2. createOrder api - Pending
3. Withdraw api - Pending
